pkg/backend/openshiftcluster_test.go

```diff
@@ -0,0 +1,399 @@
+package backend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/onsi/gomega"
+	"github.com/onsi/gomega/types"
+	uuid "github.com/satori/go.uuid"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/backend/openshiftcluster"
+	"github.com/Azure/ARO-RP/pkg/database"
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/util/billing"
+	"github.com/Azure/ARO-RP/pkg/util/deployment"
+	"github.com/Azure/ARO-RP/pkg/util/encryption"
+	mock_env "github.com/Azure/ARO-RP/pkg/util/mocks/env"
+	mock_openshiftcluster "github.com/Azure/ARO-RP/pkg/util/mocks/openshiftcluster"
+	testdb "github.com/Azure/ARO-RP/test/database"
+	testlog "github.com/Azure/ARO-RP/test/util/log"
+	teststatsd "github.com/Azure/ARO-RP/test/util/statsd"
+)
+
+type backendTestStruct struct {
+	name                            string
+	clusterDoc                      *api.OpenShiftClusterDocument
+	subscriptionDoc                 *api.SubscriptionDocument
+	mocks                           func(*mock_openshiftcluster.MockManagerInterface, database.OpenShiftClusters)
+	fixture                         func(*testdb.Fixture)
+	expectedLogs                    []map[string]types.GomegaMatcher
+	expectedDocumentExists          bool
+	expectedProvisioningState       api.ProvisioningState
+	expectedFailedProvisioningState api.ProvisioningState
+	expectedAdminUpdateError        string
+	expectedInstallPhase            api.InstallPhase
+}
+
+func TestBackendTry(t *testing.T) {
+	mockSubID := "00000000-0000-0000-0000-000000000000"
+	resourceID := fmt.Sprintf("/subscriptions/%s/resourcegroups/resourceGroup/providers/Microsoft.RedHatOpenShift/openShiftClusters/resourceName", mockSubID)
+
+	for _, tt := range []backendTestStruct{
+		{
+			name: "StateCreating with an InstallPhase set keeps it in Creating",
```

Comment:
can the first couple of test names be clarified at all?  Perhaps something like
"a queued doc in state Creating should stay in state Creating when the manager returns with properties.install set" ?

Reply:
updated a few test names